src/detect-content.c

```diff
@@ -335,6 +335,16 @@ int DetectContentSetup(DetectEngineCtx *de_ctx, Signature *s, const char *conten
     int sm_list = s->init_data->list;
     if (sm_list == DETECT_SM_LIST_NOTSET) {
         sm_list = DETECT_SM_LIST_PMATCH;
+    } else if (sm_list > DETECT_SM_LIST_MAX &&
+            0 == (cd->flags & DETECT_CONTENT_NEGATED)) {
+        /* Check transform compatibility */
+        const char *tstr;
+        if (!DetectBufferTypeValidateTransform(de_ctx, sm_list, contentstr, &tstr)) {
```

Comment:
I don't think we should evaluate the `contentstr` here, but instead `cd->content`. For example, consider these contents:
- `content:"|20|";` (match single space)
- `content:"| 00 |";` (match just a single 0 byte but note the spaces around the single byte)